cabal-install/src/Distribution/Client/Types/SourcePackageDb.hs

```diff
@@ -1,16 +1,21 @@
 {-# LANGUAGE DeriveGeneric #-}
 module Distribution.Client.Types.SourcePackageDb (
     SourcePackageDb (..),
+    lookupDependency,
+    lookupPackageName,
 ) where
 
 import Distribution.Client.Compat.Prelude
 import Prelude ()
 
 import Distribution.Types.PackageName  (PackageName)
-import Distribution.Types.VersionRange (VersionRange)
+import Distribution.Types.VersionRange (VersionRange, withinRange)
+import Distribution.Package            (packageVersion)
 
 import Distribution.Client.Types.PackageLocation (UnresolvedSourcePackage)
+import qualified Distribution.Solver.Types.PackageIndex as PackageIndex
 import Distribution.Solver.Types.PackageIndex    (PackageIndex)
+import qualified Data.Map as Map
```

Comment:
I'm not sure if we have a consistent convention about grouping imports, but I'm quite sure an import from outside our packages should be separated from imports from our package(s).